src/test/mochitest/browser_dbg-editor-gutter.js

```diff
@@ -46,4 +46,13 @@ add_task(async function() {
   await waitForDispatch(dbg, "REMOVE_BREAKPOINT");
   is(getBreakpoints(getState()).size, 0, "No breakpoints exist");
   assertEditorBreakpoint(dbg, 4, false);
+
+  // Ensure that clicking the gutter removes all breakpoints on a given line
+  await addBreakpoint(dbg, source, 4, 0);
+  await addBreakpoint(dbg, source, 4, 1);
+  await addBreakpoint(dbg, source, 4, 2);
+  clickGutter(dbg, 4);
+  await waitForDispatch(dbg, "REMOVE_BREAKPOINT", 3);
```

Comment:
I think this is one of our first tests, so it is using this older pattern of waiting for actions to fire. The problem with it is that sometimes there are timing issues around when these actions are sequenced w/ thunks. The newer style is to wait for state.

```js
await waitForState(dbg, state => getBreakpointsForLine(state, source.id, 4).size == 0)
```

it'd be nice to update the other `waitForDispatch` calls in this test as well

Reply:
Oh nice, I'll do that too.